api/signup.go

```
@@ -189,37 +226,54 @@ func (a *Api) sendSignUp(res http.ResponseWriter, req *http.Request, vars map[st
 			if a.addOrUpdateConfirmation(req.Context(), newSignUp, res) {
 				a.logMetric("signup confirmation created", req)
 
+				clinicName := "Diabetes Clinic"
+				creatorName := "Clinician"
+
+				if newSignUp.ClinicId != "" {
+					resp, err := a.clinics.GetClinicWithResponse(req.Context(), clinics.ClinicId(newSignUp.ClinicId))
+					if err != nil {
+						a.sendError(res, http.StatusInternalServerError, "unable to fetch clinic")
+						return
+					}
+					if resp.StatusCode() == http.StatusOK && resp.JSON200.Name != "" {
+						clinicName = resp.JSON200.Name
+					}
+				}
+
 				if err := a.addProfile(newSignUp); err != nil {
 					log.Printf("sendSignUp: error when adding profile [%s]", err.Error())
 					a.sendModelAsResWithStatus(res, err, http.StatusInternalServerError)
 					return
-				} else {
-					profile := &models.Profile{}
-					if err := a.seagull.GetCollection(newSignUp.UserId, "profile", a.sl.TokenProvide(), profile); err != nil {
-						a.sendError(res, http.StatusInternalServerError, STATUS_ERR_FINDING_USR, "sendSignUp: error getting user profile: ", err.Error())
-						return
-					}
+				}
+				if newSignUp.Creator.Profile != nil && newSignUp.Creator.Profile.FullName != "" {
+					creatorName = newSignUp.Creator.Profile.FullName
+				}
```

Looks like there are couple of chances for us to end up with the hard-coded defaults (e.g. if either clinic or creator name is empty). That seems unlikely scenario, but wondering if the request should proceed or fail at that point?

---

Sending invites is asynchronous and we can't guarantee the ordering of operations some operations - e.g. a clinician can delete their account before the invite is sent. Because of this, I think it's better to provide sensible defaults rather than stalling the queue (a failure in async processing in the clinic-worker acts as a poison pill).